wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -734,7 +730,15 @@ def maybe_create_imagepull_secret(
         type="kubernetes.io/dockerconfigjson",
     )
     try:
-        return core_api.create_namespaced_secret(namespace, secret)
+        try:
+            return core_api.create_namespaced_secret(namespace, secret)
+        except ApiException as e:
+            # 409 = conflict = secret already exists
+            if "409" in str(e):
```

Comment:
I think the `ApiException` has a `status` field that we can use. If "409" is in the run id, for example, we could get a false positive on this check which would produce really confusing bugs that we wouldn't be able to reproduce